module/x/peggy/spec/04_messages.md

```diff
@@ -82,83 +65,59 @@ This message will fail if:
+There are three types of confirmations a validator can submit.
 
-+++ https://github.com/althea-net/cosmos-gravity-bridge/blob/main/module/proto/peggy/v1/msgs.proto#L155-161
+- `CONFIRM_TYPE_VALSET`
+  - When the peggy daemon witnesses a complete validator set within the peggy module, the validator submits a signature of a message containing the entire validator set. 
+- `CONFIRM_TYPE_LOGIC`
+  - When a logic call request has been made, it needs to be confirmed by the bridge validators. Each validator has to submit a confirmation of the logic call being executed.
+- `CONFIRM_TYPE_BATCH`
+  - When a `MsgRequestBatch` is observed, validators need to sign batch request to signify this is not a maliciously created batch and to avoid getting slashed. 
 
-This message will fail if:
-
-- The id encoding is incorrect
-- The outgoing logic call which is confirmed can not be found
-- Invalid checkpoint generation
-- Signature decoding failed
-- The address calling this function is not a validator or its delegated key
-- The counter chain address is incorrect or empty
-- Counter party signature verification failed
-- A duplicate signature is observed
+This message is expected to fail if:
 
-### MsgDepositClaim
+- If the validator set is not present.
+- The signature is encoded incorrectly.
+- Signature verification of the ethereum key fails.
+- If the signature submitted has already been submitted previously.
+- The validator address is incorrect. 
+  - The address is empty (`""`)
+  - Not a length of 20
+  - Bech32 decoding fails
+  
+### MsgSubmitClaim
 
 When a message to deposit funds into the peggy contract is created a event will be omitted and observed a message will be submitted confirming the deposit.
 
-+++ https://github.com/althea-net/cosmos-gravity-bridge/blob/main/module/proto/peggy/v1/msgs.proto#L170-181
-
-This message will fail if:
+<!-- +++ https://github.com/althea-net/cosmos-gravity-bridge/blob/main/module/proto/peggy/v1/msgs.proto#L170-181 -->
 
-- The validator is unknown
-- The validator is not in the active set
-- If the creation of attestation fails
+There are four types of claims:
 
-### MsgWithdrawClaim
-
-When a user requests a withdrawal from the peggy contract a event will omitted by the counter party chain. This event will be observed by a bridge validator and submitted to the gravity module.
-
-
-+++ https://github.com/althea-net/cosmos-gravity-bridge/blob/main/module/proto/peggy/v1/msgs.proto#L187-193
+- `CLAIM_TYPE_DEPOSIT`
+- `CLAIM_TYPE_WITHDRAW`
+  - When a user requests a withdrawal from the peggy contract a event will omitted by the counter party chain. This event will be observed by a bridge validator and submitted to the gravity module.
+  
+- `CLAIM_TYPE_ERC20_DEPLOYED`
+  - This message allows the cosmos chain to learn information about the denom from the counter party chain.
```

Comment:
> This message allows the cosmos chain to learn information

what type of information? just the contract address and the ERC20 details (symbol, name, decimals, etc)?